src/network/networkctl.c

```diff
@@ -266,7 +267,7 @@ static int decode_netdev(sd_netlink_message *m, LinkInfo *info) {
         return 0;
 }
 
-static int decode_link(sd_netlink_message *m, LinkInfo *info, char **patterns) {
+static int decode_link(sd_netlink_message *m, LinkInfo *info, char **patterns, bool *matched_patterns) {
```

Comment:
if you write this as `bool *b` this feels as if this was only a single bool, maybe a return value or so. Would be prettier I think if we'd use [] here, to clarify this is used as array, there shouldn't be any doubts then, and C gives us the option